### DISCUSSION TITLE
Media VLC-TELNET in Music Assistant

### DISCUSSION THREAD

**Discussion by brunoblazer** [music-assistant/support#1524] [Q&A] [answered]
I installed the music assistant addon and it correctly found my google speakers, what it didn't find was the speaker that is connected via bluetooth via VLC media player via Telnet.
What is the way to bring it to the music assistant list?

**Comment by OzGav** [comment]
It has to be a supported player provider. Connect the speaker to a picoreplayer would be one way...? What is VLC running on?

**Reply by brunoblazer** [reply]
I'm using a device that passes the sound from the 3.5 jack to Bluetooth which spreads across two speakers so I didn't want to add any more equipment. i use VLC medua player via Telnet but I can use anything else as long as it transmits the sound to my bluetooth speakers

**Reply by OzGav** [reply, marked as answer]
Raspberry Pi running picoreplayer can output audio over Bluetooth https://www.picoreplayer.org/landingzone_features.html#:~:text=Stream%20from%20piCorePlayer%20to%20your%20Bluetooth%20speaker%20or%20headset.,on%20the%20piCorePlayer%20web%20interface.

**Reply by brunoblazer** [reply]
but does it require me to have an rpi dedicated to this or can I attach an integration to the HA?

**Reply by OzGav** [reply]
You could also use the squeeze lite add on for HA.

**Reply by erkr** [reply]
> You could also use the squeeze lite add on for HA. 

This solution doesn't require another PI. Just install the add on and MA can use it for the local audio. Don't install the LMS part! See https://github.com/orgs/music-assistant/discussions/1123 for the details

**Reply by brunoblazer** [reply]
Hi, I installed the music assistant, it's definitely nice but it has a big problem, it detected all the TV media or Google speakers but the output via VLC AUX for Bluetooth was not detected.

**Reply by OzGav** [reply]
You are not explaining what player provider that you expect to be detected.

**Reply by erkr** [reply]
If you expect vlc to be detected, that is not supported (see above). Two alternatives are given. Picoreplayer or squeezelite

**Reply by DutchJaFO** [reply]
I've got a bluetooth speaker attached to my HomeAssistant and using the Squeezelite add-on it does show up in MA as a Squeezelite player.

Two things to be aware of :

1. by default the speaker will be listed as a bluetooth Id (something like 'FF-AA-DD-CC'). To show a name you will need to set the name option in the squeezelite add-on. 
2. If your bluetooth speaker has been powered down for some reason then you do need to restart the squeezelite add-on to have it connect.
(check the logs of the squeezelite add-on to see it has discovered it)

From my setup :
Squeezelite add-on log showing the connected bluetooth speaker :
![image](https://github.com/music-assistant/hass-music-assistant/assets/39138434/918fe918-79fa-4b08-ba73-7f9ae7b2abce)

Music Assistant showing the speaker :
![image](https://github.com/music-assistant/hass-music-assistant/assets/39138434/d39e686a-c04a-4de9-a5b4-e8666b0a2be9)
